tests/c_api_test/test_.py

```diff
@@ -99,22 +95,27 @@ def load_from_csr(filename, reference):
         ref = reference
 
     LIB.LGBM_DatasetCreateFromCSR(
-        c_array(ctypes.c_int, csr.indptr),
-        dtype_int32,
-        c_array(ctypes.c_int, csr.indices),
-        csr.data.ctypes.data_as(ctypes.POINTER(ctypes.c_void_p)),
-        dtype_float64,
+        csr.indptr.ctypes.data_as(ctypes.POINTER(ctypes.c_int32)),
```

Comment:
could this use `.byref()` instead of `POINTER()`?

From https://docs.python.org/3/library/ctypes.html#passing-pointers-or-passing-parameters-by-reference

> ctypes exports the `byref()` function which is used to pass parameters by reference. The same effect can be achieved with the `pointer()` function, although pointer() does a lot more work since it constructs a real pointer object, so it is faster to use `byref()` if you don’t need the pointer object in Python itself.

I'm very unfamiliar with `ctypes` so apologies if that doesn't make sense.

Reply:
I wish I knew... The cited paragraph looks like we can benefit from using `byref` instead of `pointer`. I will create a new issue for this idea to not lose in the merged PR and read about it.

Reply:
Ok thanks very much!